File: docs/examples/sample.scala
package com.example.demo

import scala.collection.mutable.ListBuffer
import scala.math.sqrt

case class Point(x: Double, y: Double)

case class Circle(center: Point, radius: Double)

object MathUtils {
  def distanceBetween(a: Point, b: Point): Double = {
    val dx = a.x - b.x
    val dy = a.y - b.y
    sqrt(dx * dx + dy * dy)
  }

  def circleArea(c: Circle): Double = Math.PI * c.radius * c.radius
}

trait Shape {
  def area: Double
  def perimeter: Double
}

object Main extends App {
  val p1 = Point(0.0, 0.0)
  val p2 = Point(3.0, 4.0)
  println(s"Distance: ${MathUtils.distanceBetween(p1, p2)}")
}

File: docs/examples/sample.kt
package com.example.demo

import kotlin.collections.listOf
import kotlin.math.sqrt

data class Point(val x: Double, val y: Double)

data class Circle(val center: Point, val radius: Double)

object MathUtils {
    fun distanceBetween(a: Point, b: Point): Double {
        val dx = a.x - b.x
        val dy = a.y - b.y
        return sqrt(dx * dx + dy * dy)
    }
}

fun Circle.area(): Double = Math.PI * radius * radius

suspend fun fetchPoints(): List<Point> {
    return listOf(Point(0.0, 0.0), Point(3.0, 4.0))
}

fun main() {
    val p1 = Point(0.0, 0.0)
    val p2 = Point(3.0, 4.0)
    val c = Circle(p1, 5.0)
    println("Distance: ${MathUtils.distanceBetween(p1, p2)}")
    println("Area: ${c.area()}")
}
